src/pybind/cephfs/cephfs.pyx

```diff
@@ -1169,6 +1182,59 @@ cdef class LibCephFS(object):
                           st_atime=datetime.fromtimestamp(stx.stx_atime.tv_sec),
                           st_mtime=datetime.fromtimestamp(stx.stx_mtime.tv_sec),
                           st_ctime=datetime.fromtimestamp(stx.stx_ctime.tv_sec))
+    
+    def statx(self, path, flag):
+        """
+        Get a file's extended statistics and attributes.
+
+        :param path: the file or directory to get the statistics of.
+        :param flag: want bitfield of CEPH_STATX_* flags showing designed attributes.
```

Comment:
Looks like a good start. I'd rename "flag" parameter to "mask" to avoid confusion with the naming used in statx(2) manpage and the ceph_statx() definition in libcephfs.h.

Reply:
updated!
thanks